src/jit/morph.cpp

```diff
@@ -12739,20 +12715,13 @@ GenTree* Compiler::fgMorphSmpOpOptional(GenTreeOp* tree)
         {
             /* "x ^ -1" is "~x" */
             
-            if ((op2->gtOper == GT_CNS_INT) && (op2->gtIntConCommon.IconValue() == -1))
-            {
-                tree->ChangeOper(GT_NOT);
-                tree->gtOp2 = NULL;
-                DEBUG_DESTROY_NODE(op2);
-            }
-            else if ((op2->gtOper == GT_CNS_LNG) && (op2->gtIntConCommon.LngValue() == -1))
+            if (op2->IsIMinusOne())
```

Comment:
This is a perfect example of a case where the new abstraction fits perfectly without liberalizing the condition nor depending on the surrounding execution context.  I like these kind of changes.